packages/mdc-list/mdc-list.scss

```diff
@@ -163,25 +164,27 @@ a.mdc-list-item {
 }
 // stylelint-enable selector-max-type,selector-no-qualifying-type
 
-.mdc-list-item.mdc-ripple-upgraded {
-  @include mdc-ripple-surface;
-  @include mdc-ripple-color;
-  @include mdc-ripple-radius;
-  @include mdc-rtl-reflexive-position(left, $mdc-list-side-padding * -1);
-
+.mdc-list-item {
   position: relative;
-  // Cause the upgraded list item to cover the entirety of the list, causing ripples to emanate
-  // across the entire list element.
-  width: 100%;
-  padding: 0 $mdc-list-side-padding;
   overflow: hidden;
 
   &:focus {
     outline: none;
   }
+}
+
+// List items should support states by default, but it should be possible to opt out.
+:not(.mdc-list--non-interactive) > .mdc-list-item {
```

Comment:
Is the direct child selector (`>`) necessary to prevent the styles from leaking
into nested child `.mdc-list-item` elements?

Reply:
Without the direct child selector, this selector always matches, because while the direct parent might have the `non-interactive` modifier applied, there are plenty of ancestors that won't.